File: native/chu/rustnithm.go
package chu

type RustnithmParser struct{}

func (p *RustnithmParser) GetName() string {
	return "Rustnithm"
}

func (p *RustnithmParser) Parse(raw []byte) ChusanData {
	if len(raw) < 136 {
		return ChusanData{}
	}

	data := ChusanData{}

	for i := 0; i < 6; i++ {
		data.Air[i] = raw[i] != 0
	}

	copy(data.Slider[:], raw[6:38])

	data.Test = raw[134] != 0
	data.Service = raw[135] != 0

	return data
}

File: native/src/chu/rustnithm.rs
use super::{ChusanData, ChusanParser};

pub struct RustnithmParser;

impl ChusanParser for RustnithmParser {
    fn get_name(&self) -> &'static str {
        "Rustnithm"
    }

    fn parse(&self, raw: &[u8]) -> ChusanData {
        if raw.len() < 136 {
            return ChusanData::default();
        }

        let mut data = ChusanData::default();

        for i in 0..6 {
            data.air[i] = raw[i] != 0;
        }

        data.slider.copy_from_slice(&raw[6..38]);

        data.test = raw[134] != 0;
        data.service = raw[135] != 0;

        data
    }
}
